map_server/src/main.cpp

```diff
@@ -224,28 +264,36 @@ class MapServer
 int main(int argc, char **argv)
 {
   ros::init(argc, argv, "map_server", ros::init_options::AnonymousName);
-  if(argc != 3 && argc != 2)
+  if(argc > 3)
```

Comment:
So the allowed options are 1, 2 or 3 arguments. If 3 arguments, you get a warning. The case of 2 or 3 arguments is handled by the if (argc>1), and in case of 1 argument a server is created without loading the map.